src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -1243,36 +1238,6 @@ session_test_new (session_test_correct_t correct_client,
    return test;
 }
 
-
-static void
-check_session_returned (session_test_t *test, const bson_t *lsid)
```

Comment:
To avoid losing test coverage, could this instead pop everything from the pool, check for the `lsid`, then push it back in reverse order?